py/server/tests/test_table_factory.py

```diff
@@ -478,6 +475,48 @@ def test_j_input_wrapping(self):
         self.assertFalse(isinstance(t, InputTable))
         self.assertTrue(isinstance(t, Table))
 
+    def test_input_table_async(self):
```

Comment:
I would like to see a test or two where `on_error` gets called.  e.g. wrong schema.

Reply:
Unfortunately,  the error that cause on_error to be called isn't something we can easily produce. Added a test case to demo that.